src/Shelly.hs

```diff
@@ -1284,7 +1290,7 @@ runHandles exe args reusedHandles withHandles = do
     state <- get
 
     let cmdString = show_command exe args
-    when (sPrintCommands state) $ echo cmdString
+    when (sPrintCommands state) $ liftIO $ sPrintCommandsFn state cmdString
```

Comment:
Switching from `echo` to `sPrintCommandsFn` with standard semantics `TIO.hPutStrLn` does not seem to preserve the current behavior, see the current definition of `echo`: https://github.com/gregwebs/Shelly.hs/blob/25a7884aa5a227707968b45b1336af71c1e627f6/src/Shelly/Base.hs#L304